pkg/resources/resources.go

```diff
@@ -522,6 +522,11 @@ func Apply(ctx context.Context, cli client.Client, in client.Object, opts ...cli
 
 	err = cli.Patch(ctx, u, client.Apply, opts...)
 	if err != nil {
+		if u.GetKind() == "Secret" {
+			// For Secrets, don't log the full object or detailed error to avoid exposing sensitive data
+			return fmt.Errorf("unable to patch Secret %s/%s: operation failed", u.GetNamespace(), u.GetName())
```

Comment:
not strong opinion, but wonder if we should use this pattern for every object as I'm not sure if dumping the entire object is actually adding value